org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/shuffle/ShuffleViewer.java

```diff
@@ -0,0 +1,258 @@
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * A class for shuffling between lists
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class ShuffleViewer  {
+
+	private static final Logger logger = LoggerFactory.getLogger(ShuffleViewer.class);
+	
+	private ShuffleConfiguration conf;
+	private TableViewer fromTable, toTable;
+
+	public ShuffleViewer(ShuffleConfiguration data) {
+		this.conf = data;
+	}
+	
+	/**
+	 * Please call dispose which will clean up the listeners
+	 * and should be treated as with an SWT Displose.
+	 */
+	public void dispose() {
+		conf.clearListeners();
+	}
+	
+	public Composite createPartControl(Composite parent) {
+		
+		final Composite content = new Composite(parent, SWT.NONE);
+		content.setLayout(new GridLayout(3, false));
+		content.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+		GridUtils.removeMargins(content);
+		
+		this.fromTable = createTable(content, conf.getFromToolipText(), conf.getFromList(), "fromList", conf.getFromLabel(), false, conf.isFromReorder());		
+		createButtons(content);
+		this.toTable = createTable(content, conf.getToToolipText(), conf.getToList(), "toList", conf.getToLabel(), true, conf.isToReorder());		
+
+		return content;
+	}
+
+	private final Composite createButtons(Composite content) {
+		
+        final Composite ret = new Composite(content, SWT.NONE);
+        ret.setLayout(new GridLayout(1, false));
+        ret.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, true));
+        ret.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+        GridUtils.removeMargins(ret, 0, 20);
+
+        final Button right = new Button(ret, SWT.ARROW |SWT.RIGHT);
+        right.setToolTipText("Move item right");
+        right.setEnabled(conf.getFromList().size()>0);
+        right.addSelectionListener(new SelectionAdapter() {
+        	public void widgetSelected(SelectionEvent e) {
+        		moveRight();
+        	}
+        });
+        right.setBackground(content.getDisplay().getSystemColor(SWT.COLOR_WHITE));
+        GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+        gd.heightHint = 50;
+        right.setLayoutData(gd);
+        PropertyChangeListener pcRight = evt -> {
+			final boolean enabled = evt.getNewValue()!=null && evt.getNewValue() instanceof List && ((List)evt.getNewValue()).size()>0;
+			fromTable.getControl().getDisplay().syncExec(() -> right.setEnabled(enabled));
+        };
+        conf.addPropertyChangeListener("fromList", pcRight);
+        
+        final Button left = new Button(ret, SWT.ARROW |SWT.LEFT);
+        left.setToolTipText("Move item left");
+        left.setEnabled(conf.getToList().size()>0);
+        left.addSelectionListener(new SelectionAdapter() {
```

Comment:
If you change the signature of moveLeft can you write this as:

left.addSelectionListener(this::moveLeft);

You might not be able to do exactly that because of the adapter having two methods, but you might be able to do something a bit like that.